### ISSUE TITLE
set Form.Text in Load event cause no taskbar icon when app starting without focus

### ISSUE THREAD

**Issue by ahdung** [dotnet/winforms#9770]
### .NET version

.netfx2.0-.net6.0 tested, I believe it's ALL of .net versions.

### Did it work in .NET Framework?

No

### Issue description

Meet follow conditions, .net winform app will no icon on taskbar:
- Start without focus. for example, mouse clicks on other places during startup.
- App set `Form.Text` is in `Load` event.
- App not set `Form.Text` in `constructor` or `InitializeComponent`.

### Steps to reproduce

[NoTaskbarIconDemo.zip](https://github.com/dotnet/winforms/files/12414867/NoTaskbarIconDemo.zip)
- Compiled `exe` included.
- Note `Text` assignment in Form1.Designer.cs is commented.

https://github.com/dotnet/winforms/assets/12094247/793757e0-5fb2-4cf8-845f-b65c612ad197

**Comment by elachlan** [comment]
https://stackoverflow.com/a/16343373

There are stack overflow posts from 2009 about it. I imagine the underlying window in the winapi requires the window have a name prior to creating it.

**Comment by ahdung** [comment]
@elachlan That can't explain why no issue when assignment in `Shown` event.

**Comment by JeremyKuhne** [comment]
@Olina-Zhang can you see if you can repro this on .NET 7 or 8? I wasn't successful in my initial attempts.

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang can you see if you can repro this on .NET 7 or 8?

Yes, can also repro it on .NET7 and .NET8. Once launch application or exe, you should mouse click other place immediately with losing focus for winforms runtime. In that case, no taskbar icon of winforms runtime displays until it gets focus again.
![Net8Behavior](https://github.com/dotnet/winforms/assets/26474449/2e2aaf36-6904-4154-af6f-33bda153f4dd)

**Comment by ahdung** [comment]
@Olina-Zhang What's the screen record tool you using please?

**Comment by Olina-Zhang** [comment]
@ahdung I use ScreenToGif tool from https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/33999/How-to-record-your-screen.

**Comment by Olina-Zhang** [comment]
@ahdung or use it from https://www.screentogif.com/ if you cannot access above link.

**Comment by ahdung** [comment]
@Olina-Zhang Thank you very much.
